integration-cli/docker_cli_help_test.go

```diff
@@ -117,116 +121,29 @@ func (s *DockerSuite) TestHelpTextVerify(c *check.C) {
 		cmdsToTest = append(cmdsToTest, "volume ls")
 		cmdsToTest = append(cmdsToTest, "volume rm")
 
-		for _, cmd := range cmdsToTest {
-			var stderr string
-
-			args := strings.Split(cmd+" --help", " ")
-
-			// Check the full usage text
-			helpCmd := exec.Command(dockerBinary, args...)
-			helpCmd.Env = newEnvs
-			out, stderr, _, err = runCommandWithStdoutStderr(helpCmd)
-			c.Assert(len(stderr), checker.Equals, 0, check.Commentf("Error on %q help. non-empty stderr:%q", cmd, stderr))
-			c.Assert(out, checker.Not(checker.HasSuffix), "\n\n", check.Commentf("Should not have blank line on %q\n", cmd))
-			c.Assert(out, checker.Contains, "--help", check.Commentf("All commands should mention '--help'. Command '%v' did not.\n", cmd))
-
-			c.Assert(err, checker.IsNil, check.Commentf(out))
-
-			// Check each line for lots of stuff
-			lines := strings.Split(out, "\n")
-			for _, line := range lines {
-				c.Assert(len(line), checker.LessOrEqualThan, 107, check.Commentf("Help for %q is too long:\n%s", cmd, line))
-
-				if scanForHome && strings.Contains(line, `"`+home) {
-					c.Fatalf("Help for %q should use ~ instead of %q on:\n%s",
-						cmd, home, line)
-				}
-				i := strings.Index(line, "~")
-				if i >= 0 && i != len(line)-1 && line[i+1] != '/' {
-					c.Fatalf("Help for %q should not have used ~:\n%s", cmd, line)
-				}
-
-				// If a line starts with 4 spaces then assume someone
-				// added a multi-line description for an option and we need
-				// to flag it
-				c.Assert(line, checker.Not(checker.HasPrefix), "    ", check.Commentf("Help for %q should not have a multi-line option", cmd))
-
-				// Options should NOT end with a period
-				if strings.HasPrefix(line, "  -") && strings.HasSuffix(line, ".") {
-					c.Fatalf("Help for %q should not end with a period: %s", cmd, line)
-				}
-
-				// Options should NOT end with a space
-				c.Assert(line, checker.Not(checker.HasSuffix), " ", check.Commentf("Help for %q should not end with a space", cmd))
-
+		// Divide the list of commands to n sets and run the func testcommand on these n sets in parallel
+		// to save runtime of test
+
+		n := 5
+		perSet := len(cmdsToTest) / n
+		wg.Add(n)
+
+		for index := 0; index < n; index++ {
+			if index == n-1 {
+				go func(index int) {
+					testCommand(cmdsToTest[index*perSet:], newEnvs, scanForHome, home, c)
```

Comment:
@cpuguy83 As I am working on this, there are couple of ways to do this
1. Return from the testCommand as soon as we hit the first assert. This would require code changes to add return at every condition check. If the test condition e.g. 

     c.Assert(err, checker.IsNil, check.Commentf(out) 

is false, then return immediately. We have to add code to return at every check

2. Instead of returning at every condition check failure, collect the error in an err channel, pretend that the check is passed and continue with the remaining condition checks on this command. After we return  from the go routine testCommand, run thru the err channel and assert on (stop the test) the first error in the err channel.

I would prefer 2 as it does not require adding code to return at every check failure.